### PR TITLE
Add auto tests for property bindings

### PR DESCRIPTION
**Pull request by rakeshsadhu**

### CODE REVIEW

File: tests/propertyBinding/impl/cpp/PropertyBindingInterfaceCppImplementation.h [hunk 1]
```diff
@@ -29,21 +29,26 @@
 **********************************************************************/
 
 #pragma once
-#include "tests/propertybinding/PropertyBindingInterfaceTestImplementationBase.h"
+#include "tests/propertybinding/PropertyBindingInterfaceTestPropertyAdapter.h"
```
Comment (jacky309) [inline tests/propertyBinding/impl/cpp/PropertyBindingInterfaceCppImplementation.h]: ??

File: tests/propertyBinding/impl/qml/PropertyBindingInterfaceQmlImplementation.qml [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+** SOFTWARE.
+**
+** SPDX-License-Identifier: MIT
+**
+**********************************************************************/
+
+import QtQuick 2.0
+import tests.propertybinding 1.0
+
+PropertyBindingInterfaceTestQMLImplementation {
```
Comment (jacky309) [inline tests/propertyBinding/impl/qml/PropertyBindingInterfaceQmlImplementation.qml]: Do not use deprecated classes/components

File: tests/propertyBinding/impl/qml/PropertyBindingInterfaceQmlImplementation.qml [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+    id: root
+    Component.onCompleted: console.log("QML implementation is used.");
+
+    comboStr1: cmb
+
+    ComboStructure {
```
Comment (jacky309) [inline tests/propertyBinding/impl/qml/PropertyBindingInterfaceQmlImplementation.qml]: Why is that called Combo ?

Comment (rakeshsadhu) [reply tests/propertyBinding/impl/qml/PropertyBindingInterfaceQmlImplementation.qml]: its a structure actually, i like to call it combo as it combines few data types together

File: tests/propertyBinding/impl/qml/PropertyBindingInterfaceQmlImplementation.qml (hunk 2 above)
Comment (rakeshsadhu) [reply tests/propertyBinding/impl/qml/PropertyBindingInterfaceQmlImplementation.qml]: Changed it

File: tests/propertyBinding/impl/cpp/PropertyBindingInterfaceCppImplementation.h (hunk 1 above)
Comment (rakeshsadhu) [reply tests/propertyBinding/impl/cpp/PropertyBindingInterfaceCppImplementation.h]: @weimerb  Can answer here?

Comment (weimerb) [reply tests/propertyBinding/impl/cpp/PropertyBindingInterfaceCppImplementation.h]: Why would you change it back? Don't use the deprecated include.

Comment (rakeshsadhu) [reply tests/propertyBinding/impl/cpp/PropertyBindingInterfaceCppImplementation.h]: ok , reverted back.
Thanks

File: tests/propertyBinding/impl/qml/PropertyBindingInterfaceQmlImplementation.qml (hunk 3 above)
Comment (jacky309) [reply tests/propertyBinding/impl/qml/PropertyBindingInterfaceQmlImplementation.qml]: Combining various data types together is what a structure is all about.

Comment (rakeshsadhu) [reply tests/propertyBinding/impl/qml/PropertyBindingInterfaceQmlImplementation.qml]: so would you recommend some better name ? e.g Brend suggested TestStruct.

Comment (rakeshsadhu) [reply tests/propertyBinding/impl/qml/PropertyBindingInterfaceQmlImplementation.qml]: Changed it to PropertiesStructure And committed

Comment (rakeshsadhu) [reply tests/propertyBinding/impl/qml/PropertyBindingInterfaceQmlImplementation.qml]: @jacky309  kindly review

File: tests/propertyBinding/interface/propertyBinding.qface [hunk 4]
```diff
@@ -33,11 +33,12 @@ interface PropertyBindingInterfaceTest
...
     int intProperty2;
-    ComboStructure comboStr1;
-    ComboStructure comboStr2;
+    PropertiesStructure propertiesStruct1;
+    PropertiesStructure propertiesStruct2;
+    void setPropertiesStructureIntegerProperty(int data);
 }
 
-struct ComboStructure
+struct PropertiesStructure
```
[the remark this answers is not in the document]
Comment (rakeshsadhu) [reply tests/propertyBinding/interface/propertyBinding.qface]: done

**Comment by rakeshsadhu** [comment]
@jacky309  Can you kindly review this one?

File: tests/propertyBinding/CMakeLists.txt [hunk 5]
```diff
@@ -4,9 +4,18 @@ set(CMAKE_INCLUDE_CURRENT_DIR ON)
...
 facelift_add_qml_plugin(PropertyBindingCpp
-   URI tests.propertybinding
-   OUTPUT_BASE_DIRECTORY imports/cpp
-   SOURCES plugin/PropertyBindingPlugin.cpp
-   LINK_LIBRARIES PropertyBindingInterface
-)
+                        URI tests.propertybinding
+                        OUTPUT_BASE_DIRECTORY imports/cpp
+                        SOURCES plugin/PropertyBindingPlugin.cpp
+                        impl/cpp/PropertyBindingInterfaceCppImplementation.h
```
Comment (jacky309) [inline tests/propertyBinding/CMakeLists.txt]: please align

**Comment by rakeshsadhu** [comment]
@weimerb @jacky309  Can you review and help me in checking in ?

I see its not successfully build albeit builds on my machine with "make check" and moreover test cases which are failing have nothing to do with my check-in

File: tests/propertyBinding/CMakeLists.txt [hunk 6]
```diff
@@ -4,9 +4,18 @@ set(CMAKE_INCLUDE_CURRENT_DIR ON)
...
+    OUTPUT_BASE_DIRECTORY imports/cpp
+    SOURCES plugin/PropertyBindingPlugin.cpp
+    impl/cpp/PropertyBindingInterfaceCppImplementation.h
+    LINK_LIBRARIES PropertyBindingInterface)
+
+facelift_add_qml_plugin(PropertyBindingQml
+    URI tests.propertybinding
+    OUTPUT_BASE_DIRECTORY imports/qml
+    SOURCES plugin/PropertyBindingPlugin.cpp
+    impl/cpp/PropertyBindingInterfaceCppImplementation.h
```
Comment (jacky309) [inline tests/propertyBinding/CMakeLists.txt]: align with previous file which also belongs to SOURCES
do the same for all in this file

Comment (rakeshsadhu) [reply tests/propertyBinding/CMakeLists.txt]: done

**Pull request merged**